pyomo/gdp/plugins/bigm.py

```diff
@@ -706,3 +788,131 @@ def _estimate_M(self, expr, name):
                 M = (expr_lb, expr_ub)
 
         return tuple(M)
+
+    # These are all functions to retrieve transformed components from original
+    # ones and vice versa.
+    def get_src_disjunct(self, transBlock):
+        """Return the Disjunct object whose transformed components are on
+        transBlock.
+
+        Parameters
+        ----------
+        transBlock: _BlockData which is in the relaxedDisjuncts IndexedBlock
+                    on a transformation block.
+        """
+        if not hasattr(transBlock, '_srcDisjunct') or \
+           not type(transBlock._srcDisjunct) is weakref_ref:
```

Comment:
There are other potential issues here, right?  Like could the weakref have fallen out of scope?  Would a better trap be:
```python
try:
    return transBlock._srcDisjunct()
except:
    raise GDP_Error("Block %s doesn't appear to be a transformation "
                    "block for a disjunct. No source disjunct found." 
                    "\n\t(original error: %s)" 
                    % (transBlock.name, sys.exc_info()[1]))
```